.github/workflows/check-test-release.yml

```diff
@@ -91,6 +90,40 @@ jobs:
         pip install pre-commit .[tests]
     - if: contains(matrix.os, 'ubuntu') && matrix.python == '3.10'
       run: pre-commit run pylint -a -v --show-diff-on-failure
+    - name: Run short tests
+      timeout-minutes: 40
+      run: pytest -k 'not heroku and not flyio'
+      env:
+        GITHUB_USERNAME: ${{ secrets.GH_USERNAME }}
+        GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        GITHUB_MATRIX_OS: ${{ matrix.os }}
+        GITHUB_MATRIX_PYTHON: ${{ matrix.python }}
+        BITBUCKET_USERNAME: ${{ secrets.BITBUCKET_USERNAME }}
+        BITBUCKET_PASSWORD: ${{ secrets.BITBUCKET_PASSWORD }}
```

Comment:
IIRC, initial idea was to allow external contributors to run tests that don't require using secrets #330 #318. But here you do exactly opposite, no?

Reply:
I'm trying to recall how we ended up with this `authorize` step, and IIRC, it was the case that CI/CD doesn't run for external contributors at all because GH doesn't allow them to use secrets we put to the repo. If that's true, we need to actually separate tests to two parts: one that requires secrets, and the other that doesn't.

Reply:
I should've moved this back to draft, my bad sorry. I moved attention elsewhere but I'll catch up with you again before coming back to this.